PeoplePicker/PeoplePicker/index.ts

```diff
@@ -0,0 +1,227 @@
+    ref?: IBasePicker<IPersonaProps>;
+    defaultSelected: IPersonaProps[];
+    selectedPeople: ICustomPersonaProps[];
+    height?: number;
+    searchText: string;
+    previousSearchText: string;
+    suggestionsFilterPending?: (suggestions: IPersonaProps[]) => void;
+    refreshSuggestions: boolean;
+    resolve?: (selectedPeople: IPersonaProps[]) => void;
+    maxTags: number;
+
+    /**
+     * Used to initialize the control instance. Controls can kick off remote server calls and other initialization actions here.
+     * Data-set values are not initialized here, use updateView.
+     * @param context The entire property bag available to control via Context Object; It contains values as set up by the customizer mapped to property names defined in the manifest, as well as utility functions.
+     * @param notifyOutputChanged A callback method to alert the framework that the control has new outputs ready to be retrieved asynchronously.
+     * @param state A piece of data that persists in one session for a single user. Can be set at any point in a controls life cycle by calling 'setControlState' in the Mode interface.
+     */
+    public init(context: IPropBag<IInputs>, notifyOutputChanged: () => void): void {
+        this.notifyOutputChanged = notifyOutputChanged;
+        this.context = context;
+        this.context.mode.trackContainerResize(true);
+        this.context.parameters.Suggestions.paging.setPageSize(500);
+        this.context.parameters.Personas.paging.setPageSize(500);
+    }
+
+    onResize = (width: number, height: number): void => {
+        this.height = height;
+        this.notifyOutputChanged();
+    };
+
+    // Determining the final result by comparing it with SearchText
+    isFinalResult = (record: ComponentFramework.PropertyHelper.DataSetApi.EntityRecord): boolean => {
+        const suggestionKey = ((record.getValue(SuggestionColumns.SuggestionKey) as string) ?? '').toLowerCase();
+        const suggestionName = ((record.getValue(SuggestionColumns.SuggestionName) as string) ?? '').toLowerCase();
+        const suggestionRole = ((record.getValue(SuggestionColumns.SuggestionRole) as string) ?? '').toLowerCase();
+        const currentSearchText = this.searchText.toLowerCase();
+        return (
+            suggestionKey.indexOf(currentSearchText) > -1 ||
+            suggestionName.indexOf(currentSearchText) > -1 ||
+            suggestionRole.indexOf(currentSearchText) > -1
+        );
+    };
+
+    // To check if the Dataset is the latest by matching searchtext & To determine whether to resolve the promise
+    checkforLatestDataset = (maxRetries: number): void => {
+        let peopleListDataset = this.context.parameters.Suggestions;
+        const sortedIDArray = peopleListDataset.sortedRecordIds;
+        maxRetries === 0
+            ? //return empty result as dataset is assumed empty
+              this.returnPeople(this.resolve, true)
+            : setTimeout(() => {
```

Comment:
The use of setTimeout seems quite fragile since it only allows 5 seconds for the suggestions to be resolved - the tag picker uses the loading and updatedProperties to trigger the suggestions resolved.
What caused you to go down this setTimeout route instead?

Reply:
This was needed specifically to handle Office365Users connector based scenarios. loading and updatedProperties were not sufficient enough to mark suggestions resolved. For e.g. the suggestions(dataset) received via updateView call, with loading being false and updateProperties being true is not guaranteeing the result being correct to mark as resolve. This can be replicated in tag picker too. Agree on the point that 5 secs is fragile.